bluebrain/repo-patches/packages/steps/package.py

```diff
@@ -99,72 +100,36 @@ def patch(self):
         filter_file("unittest2", "", "requirements.txt", ignore_absent=True)
 
     def cmake_args(self):
-        args = []
-        spec = self.spec
-
-        args.append(self.define_from_variant("BUILD_TESTING", "codechecks"))
-
-        use_bundle = "ON" if "+bundle" in spec else "OFF"
-        bundles = ["EASYLOGGINGPP", "OMEGA_H", "RANDOM123", "SUNDIALS", "SUPERLU_DIST"]
-        args.extend("-DUSE_BUNDLE_{0}:BOOL={1}".format(bundle, use_bundle) for bundle in bundles)
-
-        if "+native" in spec:
-            args.append("-DTARGET_NATIVE_ARCH:BOOL=True")
-        else:
-            args.append("-DTARGET_NATIVE_ARCH:BOOL=False")
-
-        if "+lapack" in spec:
-            args.append("-DUSE_BDSYSTEM_LAPACK:BOOL=True")
-        else:
-            args.append("-DUSE_BDSYSTEM_LAPACK:BOOL=False")
-
-        if "+petsc" in spec:
-            args.append("-DUSE_PETSC:BOOL=True")
-        else:
-            args.append("-DUSE_PETSC:BOOL=False")
-
-        if "+mpi" in spec:
-            args.append("-DUSE_MPI:BOOL=True")
-        else:
-            args.append("-DUSE_MPI:BOOL=False")
-
-        if "+coverage" in spec:
-            args.append("-DENABLE_CODECOVERAGE:BOOL=True")
-
-        if "+distmesh" in spec:
-            args.append("-DSTEPS_USE_DIST_MESH:BOOL=True")
-        else:
-            args.append("-DSTEPS_USE_DIST_MESH:BOOL=False")
-
-        if "+vesicle" in spec:
-            args.append("-DSTEPS_USE_VESICLE_MODEL:BOOL=True")
-        else:
-            args.append("-DSTEPS_USE_VESICLE_MODEL:BOOL=False")
-
-        if "+stochtest" in spec:
-            args.append("-DBUILD_STOCHASTIC_TESTS:BOOL=True")
-        else:
-            args.append("-DBUILD_STOCHASTIC_TESTS:BOOL=False")
-
-        if "+codechecks" in spec:
-            args.append("-DSTEPS_TEST_FORMATTING:BOOL=ON")
-            args.append("-DSTEPS_ENABLE_ERROR_ON_WARNING:BOOL=ON")
-        else:
-            args.append("-DSTEPS_TEST_FORMATTING:BOOL=OFF")
-            args.append("-DSTEPS_ENABLE_ERROR_ON_WARNING:BOOL=OFF")
-
-        if "+caliper" in spec:
-            args.append("-DSTEPS_USE_CALIPER_PROFILING=ON")
-
-        if "+likwid" in spec:
-            args.append("-DSTEPS_USE_LIKWID_PROFILING=ON")
-
-        args.append("-DBLAS_LIBRARIES=" + spec["blas"].libs.joined(";"))
-        args.append(
-            "-DPYTHON_EXECUTABLE="
-            + spec["python"].prefix.bin.python
-            + str(spec["python"].version.up_to(1))
+        python_interpreter = self.spec["python"].prefix.bin.python + str(
+            self.spec["python"].version.up_to(1)
         )
+        args = [
+            self.define("STEPS_INSTALL_PYTHON_DEPS", False),
+            self.define("STEPS_USE_STEPSBLENDER", False),
+            self.define("USE_MPI", "True" if "+mpi" in self.spec else "False"),
+            self.define("USE_PETSC", "True" if "+petsc" in self.spec else "False"),
```

Comment:
```suggestion
            self.define_from_variant("USE_MPI", "mpi"),
            self.define_from_variant("USE_PETSC", "petsc",
```
Otherwise a thing of beauty :star_struck:

Reply:
I wish I could do that @matz-e . Unfortunately, [STEPS 4.0.0](https://github.com/CNS-OIST/HBP_STEPS/blob/4.0.0/CMakeLists.txt#L250) requires a string with values "Default", "True", or "False", whereas the `define_from_variant` sets value ON/OFF. This is very old code...

Reply:
OK